### PR TITLE
Interfaces

### PR DESCRIPTION
**Pull request by chrispduck**
* Initialise the hardhat project
* Create an interface for the pawnshop
* Update the Pawnshop to be compatible with the interface

### CODE REVIEW

File: contracts/interfaces/IPawnshop.sol [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    // statusUpdater is called by the platform to keep listings
+    // up to date @Admin? TODO
+    function singleStatusUpdater(uint256 _lendingid) external;
+
+    // buy is called by a buyer to purchase an asset which is for sale
+    function buy(uint256 _lendingId) external payable;
+
+    // pay is called by the lender to return the capital.
+    // The value paid must be exactly to the amount owed.
+    function pay(uint256 _lendingId) external payable;
```
Comment (chrispduck) [inline contracts/interfaces/IPawnshop.sol:60]: maybe this should be called `repay`

File: contracts/interfaces/IPawnshop.sol [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    event FullyFunded (uint256 lendingId);
+    // Emitted when the review-time of a lending expired with insufficient funding.
+    event InsufficientFunding (uint256 lendingId);
+    // Emitted when an asset is available to purchase
+    event ForSale (uint256 lendingId);
+    // Emitted when an asset is sold
+    event Sold(address buyer, uint256 lendingId, uint256 amount);
+
+    // Set the interest rate in the contract. Used in future lendings.
+    // @Admin? TODO
```
Comment (chrispduck) [inline contracts/interfaces/IPawnshop.sol:17]: We could potentially split off Admin functions to a separate interface. Making it clearer which functions are for users

**Comment by chrispduck** [comment]
The methods which take Lending as an argument or return value are hard to capture in an interface. I have omitted them for now.

File: test/sample-test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    expect(await greeter.greet()).to.equal("Hello, world!");
+
+    const setGreetingTx = await greeter.setGreeting("Hola, mundo!");
+
+    // wait until the transaction is mined
+    await setGreetingTx.wait();
+
+    expect(await greeter.greet()).to.equal("Hola, mundo!");
+  });
+});
```
Comment (anajuliabit) [inline test/sample-test.js]: I think we don't need this example file

File: scripts/sample-script.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+// We recommend this pattern to be able to use async/await everywhere
+// and properly handle errors.
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
```
Comment (anajuliabit) [inline scripts/sample-script.js]: Same here, I think we don't need this file

File: contracts/interfaces/IPawnshop.sol [hunk 5]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+interface IPawnshop {
+    // Emitteed when a borrower calls borrow
```
Comment (anajuliabit) [inline contracts/interfaces/IPawnshop.sol:5]: We can use natspec format for comments. What do you think?

**Review by anajuliabit — CHANGES_REQUESTED** [review]
Awesome work here @chrispduck!

Just a few suggestion:

File: contracts/interfaces/IPawnshop.sol (hunk 5 above)
Comment (chrispduck) [reply contracts/interfaces/IPawnshop.sol:5]: Yeah! Happy to natspec the comments. Would be good to get feedback from @Constantino on the interface first before spending time prettying

File: test/sample-test.js (hunk 3 above)
Comment (chrispduck) [reply test/sample-test.js]: Agreed. Removed :)

File: scripts/sample-script.js (hunk 4 above)
Comment (chrispduck) [reply scripts/sample-script.js]: Deleted, thanks

**Pull request merged**